burst/src/main/java/com/squareup/burst/annotation/Name.java

```diff
@@ -0,0 +1,44 @@
+package com.squareup.burst.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Marks a field or parameter name for test output. This can help make test output more
+ * readable.
+ * <p>
+ * The following snippet:
+ * <pre>
+ *   package com.example.yourpackage
+ *
+ *   public class MyTests {
+ *     enum BooleanCondition { TRUE, FALSE }
+ *
+ *     {@literal @}Name("IsEnabled") private BooleanCondition isEnabled;
+ *     {@literal @}Name("IsFlagged") private BooleanCondition isFlagged;
+ *
+ *     …
+ *
+ *     {@literal @}Test
+ *     public void testConditions() {
+ *       …
+ *     }
+ *   }
+ * </pre>
+ * Would produce output like this in the runner:
+ * <pre>
+ *   testCondition[](com.example.yourpackage.MyTest[IsEnabled.TRUE, IsFlagged.TRUE])
+ *   testCondition[](com.example.yourpackage.MyTest[IsEnabled.TRUE, IsFlagged.FALSE])
+ *   testCondition[](com.example.yourpackage.MyTest[IsEnabled.FALSE, IsFlagged.TRUE])
+ *   testCondition[](com.example.yourpackage.MyTest[IsEnabled.FALSE, IsFlagged.FALSE])
+ * </pre>
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.FIELD, ElementType.PARAMETER })
+public @interface Name {
+  String EMPTY = "";
+
+  String value() default EMPTY;
```

Comment:
I was thinking in the event that they don't want the name at all. Agreed that it's not really helpful to have that as the default though, I'll remove it.